source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,797 @@
+    ( ( !defined( FSL_SDK_DISBLE_L2CACHE_PRESENT ) ) || ( FSL_SDK_DISBLE_L2CACHE_PRESENT == 0 ) )
+        #if defined( FSL_FEATURE_L1DCACHE_LINESIZE_BYTE )
+            #define FSL_CACHE_LINESIZE_MAX     MAX( FSL_FEATURE_L1DCACHE_LINESIZE_BYTE, FSL_FEATURE_L2CACHE_LINESIZE_BYTE )
+            #define FSL_ENET_BUFF_ALIGNMENT    MAX( ENET_BUFF_ALIGNMENT, FSL_CACHE_LINESIZE_MAX )
+        #else
+            #define FSL_ENET_BUFF_ALIGNMENT    MAX( ENET_BUFF_ALIGNMENT, FSL_FEATURE_L2CACHE_LINESIZE_BYTE )
+        #endif
+    #elif defined( FSL_FEATURE_L1DCACHE_LINESIZE_BYTE )
+        #define FSL_ENET_BUFF_ALIGNMENT        MAX( ENET_BUFF_ALIGNMENT, FSL_FEATURE_L1DCACHE_LINESIZE_BYTE )
+    #else
+        #define FSL_ENET_BUFF_ALIGNMENT        ENET_BUFF_ALIGNMENT
+    #endif /* if defined( FSL_FEATURE_L2CACHE_LINESIZE_BYTE ) && ( ( !defined( FSL_SDK_DISBLE_L2CACHE_PRESENT ) ) || ( FSL_SDK_DISBLE_L2CACHE_PRESENT == 0 ) ) */
+#else /* if defined( FSL_SDK_ENABLE_DRIVER_CACHE_CONTROL ) && FSL_SDK_ENABLE_DRIVER_CACHE_CONTROL */
+    #define FSL_ENET_BUFF_ALIGNMENT            ENET_BUFF_ALIGNMENT
+#endif /* if defined( FSL_SDK_ENABLE_DRIVER_CACHE_CONTROL ) && FSL_SDK_ENABLE_DRIVER_CACHE_CONTROL */
+
+/* A bigger value is chosen so that the previous notification values are
+ * not interfering with the driver ready notifications. */
+#define DRIVER_READY    ( 0x40 )
+#define DRIVER_FATAL    ( DRIVER_READY << 1 )
+
+typedef uint8_t   rx_buffer_t[ SDK_SIZEALIGN( ENET_RXBUFF_SIZE, FSL_ENET_BUFF_ALIGNMENT ) ];
+typedef uint8_t   tx_buffer_t[ SDK_SIZEALIGN( ENET_TXBUFF_SIZE, FSL_ENET_BUFF_ALIGNMENT ) ];
+
+/**
+ * @brief Used to wrap received data in a pbuf to be passed into lwIP
+ *        without copying.
+ * Once last reference is released, buffer can be used by ENET RX DMA again.
+ */
+typedef struct rx_pbuf_wrapper
+{
+    void * buffer;             /*!< Original buffer wrapped. */
+    volatile bool buffer_used; /*!< Wrapped buffer is used by ENET or FreeRTOS+TCP. */
+} rx_pbuf_wrapper_t;
+
+/**
+ * Helper structure to hold private data used to operate your Ethernet interface.
+ */
+struct ethernetif
+{
+    ENET_Type * base;
+    enet_handle_t handle;
+    enet_rx_bd_struct_t * RxBuffDescrip;
+    enet_tx_bd_struct_t * TxBuffDescrip;
+    rx_buffer_t * RxDataBuff;
+    tx_buffer_t * TxDataBuff;
+    rx_pbuf_wrapper_t RxPbufs[ ENET_RXBUFF_NUM ];
+};
+
+typedef enum xEMAC_STATE
+{
+    xEMAC_SetupPHY,
+    xEMAC_WaitPHY,
+    xEMAC_Init,
+    xEMAC_Ready,
+} EMACState_t;
+
+static EMACState_t eEMACState = xEMAC_SetupPHY;
+
+static mdio_handle_t mdioHandle = { .ops = &EXAMPLE_MDIO_OPS };
+
+static phy_handle_t phyHandle = { .phyAddr = 0x00, .mdioHandle = &mdioHandle, .ops = &EXAMPLE_PHY_OPS };
+
+/**
+ * The task-handle for deferred interrupt handler task that processes
+ * incoming packets.
+ */
+static TaskHandle_t receiveTaskHandle = NULL;
+
+static struct ethernetif EthernetInterface1;
+struct ethernetif * ethernetifLocal = &EthernetInterface1;
```

Comment:
Is `ethernetifLocal` used by NXP ethernet driver or anywhere outside, if not, can we make it `static`?

Reply:
Why copy it to a local variable?